fennel/datasets/datasets.py

```diff
@@ -853,16 +853,27 @@ def dsschema(self):
 
 
 class Dedup(_Node):
-    def __init__(self, node: _Node, by: List[str]):
+    def __init__(self, node: _Node, by: List[str], window: Optional[Union[Session, Tumbling]] = None):
         super().__init__()
         self.node = node
         self.by = by
         self.node.out_edges.append(self)
-
+        self.window = window
+        
+        if window is not None:
+            if not isinstance(window, (Session, Tumbling)):
+                raise TypeError(
+                    "Type of 'window' param can only be either Session or Tumbling."
```

Comment:
Let's also reference that the error is in dedup operator:

`invalid dedup operator: 'window' can either be Session or Tumbling but found 'xyz'`